### ISSUE TITLE
What WCAG rules does it support?

### ISSUE THREAD

**Issue by fracz** [yargalot/AccessSniff#23]
I am using your lib with a gulp plugin and the following code:

```
gulp.src('index.html')
        .pipe(accessibility({
            accessibilityLevel: 'WCAG2AAA',
            ignore: [
                // ignored, because I want to lint partial html files, not the whole pages
                'WCAG2AAA.Principle2.Guideline2_4.2_4_2.H25.1.NoTitleEl',
                'WCAG2AAA.Principle3.Guideline3_1.3_1_1.H57.2'
            ]
        }))
```

Having the following index.html:

```
<img src="whatever.png">
```

All is says is that `1 file lint free!`. Hovewer, I am pretty sure WCAG specifies there must be an alternative text for all images (see [the very first rule](https://www.w3.org/TR/WCAG20/#text-equiv)).

So my quesiton is what rules does this plugin really support?

**Comment by yargalot** [comment]
I imagine it supports all the rules, however thats an odd one. The base is built off HTML Codesniffer so might be a big with that. Can you post up some files/example code so I can look into it further.

**Comment by JCron245** [comment]
I tested this out with an HTML file that just included the one line of code
```
<img src="someplace.jpg">
```

However looking at your code, and the Gulp task I used:
```
//Make a WCAG report output to ./reports/wcag_report.txt
gulp.task('wcag_report', function() {
  gulp.src(html_paths)
  .pipe(access({
    force: true,
    accessibilityLevel: 'WCAG2AAA',
    verbose: false,
    ignore: [
    // ignored, because I want to lint partial html files, not the whole pages
    	'WCAG2AAA.Principle2.Guideline2_4.2_4_2.H25.1.NoTitleEl',
        'WCAG2AAA.Principle3.Guideline3_1.3_1_1.H57.2'
    ]
  }))
  .pipe(access.report({reportType: 'csv'}))
  .pipe(rename({
    verbose: false,
    prefix: 'wcag',
    basename: '-report',
    extname: '.csv'
  }))
  .pipe(gulp.dest('reports/'));
});
```
I noticed that you are not outputting errors to a file or the console, which would allow you to see the errors. When running this task when it is finished it always outputs `1 file lint free!`. I tested it by outputting to a file and to the cli and got the expected output.
Specifically my output was:
```
heading, issue, element, id, class, line, column, description 
ERROR,"WCAG2AAA.Principle1.Guideline1_1.1_1_1.H37",<img src="someplace.jpg">,,,1,0,Img element missing an alt attribute. Use the alt attribute to specify a short text alternative.

NOTICE,"WCAG2AAA.Principle1.Guideline1_1.1_1_1.G73,G74",<img src="someplace.jpg">,,,1,0,If this image cannot be fully described in a short text alternative, ensure a long text alternative is also available, such as in the body text or through a link.
```

I believe your problem is that you need to either output to the console with something like
`.on('error', console.log);` or by setting up a file to output the results to that way you can see the actual warnings/errors/notices

**Comment by JCron245** [comment]
And to answer the overall question it appears to support all the WCAG2(A,AA,AAA) rules, I have done quite a bit of testing of those and it has caught everything I have thrown at it so far.

**Comment by fracz** [comment]
When I use the plugin without the gulp plugin and call it with path that directly indicates the file (e.g. `AccessSniff.default(['files/sample.html'])...`), it detects the missing `alt` correctly.

However, if I use a wildcard e.g. `AccessSniff.default(['files/*.html'])...`, the missing `alt` is not detected.

I have created a [minimal example](https://github.com/fracz/accesssniff-test).

It behaves as follows:

```
$ npm install
...
$ node this-detects-missing-alt.js
Starting Accessibility tests
Testing files/sample.html
Tested files/sample.html

ERROR WCAG2A.Principle1.Guideline1_1.1_1_1.H37
Line:1 Col:0
Img element missing an alt attribute. Use the alt attribute to specify a short text alternative.
--------------------
<img src="whatever.png">

NOTICE WCAG2A.Principle1.Guideline1_1.1_1_1.G73,G74
Line:1 Col:0
If this image cannot be fully described in a short text alternative, ensure a long text alternative is also available, such as in the body text or through a link.
--------------------
<img src="whatever.png">

1 file lint free!
Writing JSON Report...

$ node this-does-not-detect-missing-alt.js
Starting Accessibility tests
Testing files/*.html
Writing JSON Report...
```

Could you try if it behaves the same way for you?

**Comment by yargalot** [comment]
Looks like it was a globbing issue :) Tried it out on the new repo and seems to work

**Comment by fracz** [comment]
Thank you very much, works now :-)
